app/scripts/metamask-controller.js

```diff
@@ -200,7 +200,7 @@ module.exports = class MetamaskController extends EventEmitter {
       pluginAccountsController: this.pluginAccountsController,
     }, initState.AccountsController)
     this.accountsController.store.subscribe((s) => this._onAccountControllerUpdate(s))
-
+    this.keyringController.memStore.subscribe((s) => this._onKeyringControllerUpdate(s))
```

Comment:
There is probably a refactor opportunity in the future, to wean metamask-controller off the keyring controller.